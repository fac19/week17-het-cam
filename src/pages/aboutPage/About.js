import React from "react";
import BackIcon from "../../components/global/icons/back/BackIcon";
import Heading from "../../components/global/heading/Heading";
import { TextButton } from "../../components/global/buttons/Buttons";
import { Text, AboutWrapper } from "./About.style";
import netlifyIdentity from "netlify-identity-widget";
import { useHistory } from "react-router-dom";

const About = () => {
  const history = useHistory();
  return (
    <>
      <BackIcon to={"/"} />
      <Heading>About the App</Heading>
      <AboutWrapper>
        <Text>Welcome to Enterprise Adventure.</Text>
        <Text>
          This brand new app will take you on a life-changing journey!
        </Text>
        <Text>
          Take part in exciting missions to score points, level up and collect
          badges.
        </Text>
        <Text>
          Along the way you’ll develop essential skills to succeed in school,
          work and life.
        </Text>
        <Text>Don’t delay – sign up now!</Text>
        <TextButton
          type={"button"}
          text={"Sign Up and Join the Movement!"}
          onClick={() => {
            netlifyIdentity.open();
            history.push(`/signup`);
          }}
        />
      </AboutWrapper>
    </>
  );
};

export default About;
