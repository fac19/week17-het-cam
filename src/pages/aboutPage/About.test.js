import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import About from "./About";

test("Landing page component is styled", () => {
  const tree = renderer
    .create(<About type={"button"} text={"Sign Up and Joing the Movement!"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Sign up button renders correctly", () => {
  render(<About type={"button"} text={"Sign Up and Joing the Movement!"} />);
  screen.getByRole("heading");
  screen.getByText("About the App");
});
