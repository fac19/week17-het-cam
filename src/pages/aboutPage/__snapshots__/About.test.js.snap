// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component is styled 1`] = `
Array [
  .c0 {
  border-radius: 100%;
  border: none;
  background-color: #026CB6;
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
}

@media (min-width:600px) {

}

<button
    aria-label="Go back to previous page"
    className="c0"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-root-1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
  </button>,
  .c0 {
  font-family: "Josefin Sans",sans-serif;
  font-size: 2rem;
  color: #032049;
  text-align: center;
}

@media (min-width:600px) {
  .c0 {
    font-size: 4rem;
  }
}

<h1
    className="c0"
  >
    About the App
  </h1>,
  .c1 {
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:600px) {

}

<section
    className="c0"
  >
    <p
      className="c1"
    >
      Welcome to Enterprise Adventure.
    </p>
    <p
      className="c1"
    >
      This brand new app will take you on a life-changing journey!
    </p>
    <p
      className="c1"
    >
      Take part in exciting missions to score points, level up and collect badges.
    </p>
    <p
      className="c1"
    >
      Along the way you’ll develop essential skills to succeed in school, work and life.
    </p>
    <p
      className="c1"
    >
      Don’t delay – sign up now!
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-2"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sign Up and Join the Movement!
      </span>
    </button>
  </section>,
]
`;
